### DISCUSSION TITLE
A routes-manifest.json couldn't be found

### DISCUSSION THREAD

**Discussion by willTastyMakers** [vercel/vercel#4213] [General] [answered]
I have spent the past 2 days trying to deploy with vercel. It used to be fairly straight forward now I can't make heads or tells of why I am getting this
```
Error: A routes-manifest.json couldn't be found. This could be due to a failure during the build
13:04:35.457  
    at Object.getRoutesManifest (/zeit/1bc6bc026129a8fb/.build-utils/.builder/node_modules/@now/next/dist/index.js:6113:15)
13:04:35.457  
    at async Object.module.exports.178.exports.build (/zeit/1bc6bc026129a8fb/.build-utils/.builder/node_modules/@now/next/dist/index.js:3764:28)
13:04:35.457  
    at async buildStep (/var/task/sandbox-worker.js:20373:20)
13:04:35.457  
    at async mainSub (/var/task/sandbox-worker.js:20118:9)
13:04:35.457  
    at async main (/var/task/sandbox-worker.js:20023:5)
Reviewing similar issues on github...
```

Searching other issues and I know this are not related to my repo
don't have .now directory
do have a public directory

searched your docs for Routes Manifest Definition and found nothing

**Comment by coetry** [comment]
Hi @willTastyMakers can you give us some context on what you're trying to deploy? Is this a Next.js app following a conventional directory structure or does are you perhaps trying to deploy a Next app from a different subdirectory?

Do you have a repo or a reproducible example for us?

**Reply by willTastyMakers** [reply]
Hi Coetry,
Thanks for getting back to me.  It's a private repo for client, so I can not expose in its current form, but perhaps I can explain what is happening.


I have created an application that has 3 pages.
// pages
**index.js** 
( home page, that should be render as a static Optimized html)

**login.js, clientPortal.js** 
( are rendered on the server, they are not slug type pages [slug].js, but they both are expected to use the server rendering ( getInitialProps). The pages use the getInitialProps to check for the existence of a **cookie/JWTtoken**. If the token is present an api call is made to a server in a cloud service (GCP). If the user clicks on  Login link, from the Splash page (index.js), they will be redirected to the clientPortal.js page if the cookie is valid, if not then they will land on the login.js page.)

I don't understand what I am missing here. Everything seems to work accordingly on my local client setup but continually fails when I deploy. The error I am receiving is 
```
Error: A routes-manifest.json couldn't be found. This could be due to a failure during the build
13:04:35.457  
```
I am not using a  now.json, if I do need to use one in order to place routes for clientPortal and login, how should that work.

Thank you, ( I have been banging away for the past 10 hours looking for a solution(

**Comment by willTastyMakers** [comment]
Hi  @coetry ,
Your mate, @samsisle (samuel ko), was able to help me. Turns out the issue was merely a '/public'  that I added in the Vercel Dashboard settings input field for the Output Directory input . I thought I need to do this so Vercel could find my code, because it was initially just loading a view of my directories in the now.sh view. Once the ./public was removed everything worked as expected.

Cheers.
Bro

**Comment by styfle** [comment, marked as answer]
This issue has been fixed in PR #5009, thanks!
